### PR TITLE
Add logic to prevent column collapse

### PR DESCRIPTION
**Pull request by Suvra19**

### CODE REVIEW

File: src/actions/ResizeTableActions.js [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
 
     let visibleColumnsWidth = 0;
     visibleColumns.map(column => visibleColumnsWidth += column.minWidth);
 
-    state = visibleColumnsWidth > width ?
-        tryToRemoveColumns({ visibleColumnsWidth, width, state }) :
-        tryToAddColumns({ visibleColumnsWidth, width, state });
-
+    if (isCollapsible) {
+        state = visibleColumnsWidth > width || !isCollapsible ?
```
Comment (JonForest) [inline src/actions/ResizeTableActions.js]: We already know that `isCollapsible` is `true` here, as you've checked it in the line above.

File: src/actions/ResizeTableActions.js [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
     return state;
 };
 
+// Adds all hidden columns to the state irrespective of the available width
+export const addAllColumns = ({ state }) => {
+    const { columns } = state;
+    let hiddenColumns = Object.assign([], columns.filter(column => !column.isVisible ));
+    hiddenColumns.sort(dynamicSort({column: 'priorityLevel'}));
+    
+    while(hiddenColumns.length !== 0){
```
Comment (JonForest) [inline src/actions/ResizeTableActions.js]: Spacing needs adding here to be consistent.

File: src/actions/ResizeTableActions.js [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
+        state = addAllColumns({ state });
+    }
+    
     return state;
 };
 
+// Adds all hidden columns to the state irrespective of the available width
+export const addAllColumns = ({ state }) => {
+    const { columns } = state;
+    let hiddenColumns = Object.assign([], columns.filter(column => !column.isVisible ));
```
Comment (JonForest) [inline src/actions/ResizeTableActions.js]: You never re-assigned `hiddenColumns`, so it can be `const`

**Review by JonForest — APPROVED** [review]
LGTM.  But please merge following the other PR.

File: src/actions/ResizeTableActions.js (hunk 1 above)
Comment (Suvra19) [reply src/actions/ResizeTableActions.js]: Yup, good catch. I forgot to remove that.

File: src/actions/ResizeTableActions.js (hunk 2 above)
Comment (Suvra19) [reply src/actions/ResizeTableActions.js]: Done!

File: src/actions/ResizeTableActions.js (hunk 3 above)
Comment (Suvra19) [reply src/actions/ResizeTableActions.js]: Correct. Changed it to ``const``

**Pull request merged**